### ISSUE TITLE
Use amp-list with Google Place Autocomplete API

### ISSUE THREAD

**Issue by sryze** [ampproject/amphtml#13527]
## What's the issue?

It doesn't seem to be possible to use the [Place Autocomplete API](https://developers.google.com/places/web-service/autocomplete) with `amp-list`:

> No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8080' is therefore not allowed access. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
> Error fetching amp-list: XHR Failed fetching (https://maps.googleapis.com/...): Failed to fetch​​​

Is there a workaround?

I would like to implement an auto suggestions feature where when a user types a city name a list of cities is displayed below the input. I haven't been able to find an AMP component for Google Places, which is  kind of strange because this is also a Google project.

Should I use an iframe instead? I can't image how it would be done with an iframe, since content can't overflow outside of iframe's bounds I would have to allocate space for the dropdown list in advance. The page that I'm trying to convert to AMP basically has this input at the very top, it's the entry point to the site.

## How do we reproduce the issue?

```javascript
<amp-list width="auto"
	height="100"
	layout="fixed-height"
	src="https://maps.googleapis.com/maps/api/place/autocomplete/json?input=new york&key=YOUR_KEY_HERE&types=(cities)&components=country:us">
	<template type="amp-mustache">
	</template>
</amp-list>
```

**Comment by sryze** [comment]
I've found a way to implement this using `amp-list` and a bunch of other componets. For those who are interested see this example:

https://ampbyexample.com/advanced/autosuggest

Note: You probably won't be able to call the Place Autocomplete API directly from the browser using `<amp-list>` because of CORS. I worked around this by implementing an endpoint in my server that calls Google's API. I don't know if there is a better way, but this worked for me.
